src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -34,11 +34,25 @@ public static class Helpers
                 "void", "partial", "yield", "where"
             };
 
+        private static int numSafeIden = 0;
         public static string SafeIdentifier(string id)
         {
             if (id.All(char.IsLetterOrDigit))
                 return Keywords.Contains(id) ? "@" + id : id;
-            return new string(id.Select(c => char.IsLetterOrDigit(c) ? c : '_').ToArray());
+            
+            id = new string(id.Select(c => char.IsLetterOrDigit(c) ? c : '_').ToArray());
+
+            if (id.Replace("_", "") == string.Empty)
+            {
+                id = string.Empty;
+                ++numSafeIden;
+                for (var i = 0; i < numSafeIden; ++i)
+                {
+                    id += '_';
```

Comment:
Never concatenate ("+") strings in a loop. Creating strings is an expensive operation because strings are immutable which means that each concatenation creates a new string object. Use StringBuilder.Append instead.